src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,392 @@
+            {
+                Add(property);
+            }
+        }
+
+        /// <summary>
+        ///   Removes the property with the specified name.
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public bool Remove(string propertyName) => _dictionary.Remove(propertyName);
+
+        /// <summary>
+        ///   Determines whether a property is in a JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to check.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is found in a JSON object,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public bool ContainsProperty(string propertyName) => _dictionary.ContainsKey(propertyName);
+
+        /// <summary>
+        ///   Modifies the name of specified property.
+        /// </summary>
+        /// <param name="oldName">Old name of the property to change.</param>
+        /// <param name="newName">New name of the property to change.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        public void ModifyPropertyName(string oldName, string newName)
+        {
+            if (!_dictionary.ContainsKey(oldName))
+                throw new KeyNotFoundException(SR.PropertyNotFound);
+
+            JsonNode previousValue = _dictionary[oldName];
+            _dictionary.Remove(oldName);
+            _dictionary.Add(newName, previousValue);
+        }
+
+        /// <summary>
+        ///   Returns the property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <returns>Value of the property with the specified name.</returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        public JsonNode GetProperty(string propertyName)
+        {
+            if (!TryGetProperty(propertyName, out JsonNode jsonNode))
+            {
+                throw new KeyNotFoundException(SR.PropertyNotFound);
+            }
+
+            return jsonNode;
+        }
+
+        /// <summary>
+        ///   Returns the property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <param name="jsonNode">Value of the property with specified name.</param>
+        /// <returns>
+        ///  <see langword="true"/> if property with specified name was found;
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
```

Comment:
Remark that when `TryGetProperty` returns false, JsonNode is meaningless. Return `JsonNode` as null doesn't mean, the property value was "null" (unless TryGetProperty returns true).

Reply:
Oh, yes, I see, thanks for pointing this out!